test/macros/spawn_lnd_daemon.js

```diff
@@ -0,0 +1,235 @@
+        ];
+        break;
+      default:
+        return cbk(['0', 'UnknownLightningBackendChain']);
+      }
+      let lndDaemon;
+      lndDaemon = spawn('lnd', lndParams);
+      lndDaemon.stderr.on('err', data => { });
+      let manualKill = false;
+      lndDaemon.stdout.on('data', data => {
+          if (namespaceFail.test(`${data}`)) {
+            lndDaemon.kill();
+            manualKill = true;
+
+            lndDaemon = spawn('lnd', lndParams);
+
+            lndDaemon.stdout.on('data', data => {
+              if (rpcServerReady.test(`${data}`)) {
+                setTimeout(function () {
+                  return cbk(null, {is_ready: true, lndDir});
+                }, 2000);
+              }
+            });
+
+            lndDaemon.on('close', code => {
+              removeDir(lndDir, () => {});
+              removeDir(validateCredentials.lndChainDir, () => {});
+            });
+
+            if (rpcServerReady.test(`${data}`)) {
+              setTimeout(function () {
+                return cbk(null, {is_ready: true, lndDir});
+              }, 2000);
+            }
+          }
+        }
+      );
+
+      lndDaemon.on('close', code => {
+        if (!manualKill) {
+          removeDir(lndDir, () => {});
+          removeDir(validateCredentials.lndChainDir, () => {});
+        }
+      });
+
+      process.on('uncaughtException', err => {          // daemon.kill();
+        process.exit(1);
+      });
+
+    }],
+
+    spawnLNDInterface: ['spawnLND', ({spawnLND}, cbk) => {
+      if (!spawnLND.is_ready) {
+        return cbk([0, "LNDaemon failed to ready"]);
+      }
+      const cert = new Buffer(fs.readFileSync(path.join(spawnLND.lndDir, 'tls.cert'))).toString('base64');
+      const macaroon = new Buffer(fs.readFileSync(path.join(spawnLND.lndDir, 'admin.macaroon'))).toString('base64');
+      const host = '127.0.0.1:10009';
+      const lnd = lightningDaemon({cert, host, macaroon});
+      return cbk(null, {lnd});
+
+    }],
+
+    verifyLNDInterface: ['spawnLNDInterface', ({spawnLNDInterface}, cbk) => {
+      signMessage({lnd: spawnLNDInterface.lnd, message: " "}, (err, res) => {
+        if (!res.signature) {
+          return cbk([0, 'ExpectedValidLNDSignedMessage']);
+        } else {
+          setTimeout(() => {
+            return cbk(null, {lnd: spawnLNDInterface.lnd});
+          }, 5000);
+        }
+      });
+    }],
+
+    genAddress: ['verifyLNDInterface', ({verifyLNDInterface}, cbk) => {
+      return createAddress({lnd: verifyLNDInterface.lnd}, cbk);
+    }],
+
+    startNoTLSBTCD: ['genAddress', 'validateCredentials', ({genAddress, validateCredentials}, cbk) => {
+      spawnChainDaemon({
+        network: args.network,
+        daemon: "btcd",
+        dir: validateCredentials.baseChainDir,
+        finished_mining_public_key: genAddress.address,
+        simnet: true,
+        tls: false,
+      }, (err, res) => {
+        if (err) {
```

Comment:
Style for single errs is they don't need an else

Reply:
fixed